src/app/COMPONENTS/task-details/task-details.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit } from '@angular/core'
+import { Task } from '../../models/task.model'
+import { MaterialModule } from '../../material.module'
+import { TasksService } from '../../service/tasks.service'
+import { ActivatedRoute } from '@angular/router'
+import { Observable, switchMap, take } from 'rxjs'
+import { CommonModule, NgIf } from '@angular/common'
+import { FormBuilder, FormControl, Validators, ReactiveFormsModule, FormGroup } from '@angular/forms'
+
+@Component({
+  selector: 'app-task-details',
+  standalone: true,
+  imports: [
+    MaterialModule,
+    CommonModule,
+    NgIf,
+    ReactiveFormsModule
+  ],
+  templateUrl: './task-details.component.html',
+  styleUrl: './task-details.component.scss'
+})
+export class TaskDetailsComponent implements OnInit {
+  
+  taskForm: FormGroup = this.fb.group({
+    title: new FormControl({ value: "", disabled: true }, Validators.required),
+    description: new FormControl({ value: "", disabled: true }, Validators.required),
+    type: new FormControl({ value: "", disabled: true }, Validators.required),
+    status: new FormControl({ value: "", disabled: true }, Validators.required),
+    createdOn: new FormControl({ value: "", disabled: true }, Validators.required),
+  })
+  
+  task$: Observable<Task> | null = null
+
+  constructor(private tasksService: TasksService, private activatedRoute: ActivatedRoute, private fb: FormBuilder) { }
+
+  ngOnInit() {
+    this.activatedRoute.params.pipe(switchMap((data) => {
```

Comment:
instead of using switchMap at first, you should use the map operator to find and return only the id param (string) after that you use the switchMap operator to take that ID and return the task, that way you will be returning a type Observable<Task> which you can assign directly to the task$ variable